x-pack/plugins/ml/server/lib/alerts/jobs_health_service.ts

```diff
@@ -0,0 +1,173 @@
+
+    const includeAllJobs = jobAndGroupIds.some((id) => id === ALL_JOBS_SELECTION);
+
+    // Extract jobs from group ids and make sure provided jobs assigned to a current space
+    const jobsResponse = (
+      await mlClient.getJobs<MlJobsResponse>({
+        ...(includeAllJobs ? {} : { job_id: jobAndGroupIds.join(',') }),
+      })
+    ).body.jobs;
+
+    let resultJobIds = jobsResponse.map((v) => v.job_id);
+
+    if (excludeJobs && (!!excludeJobs.jobIds.length || !!excludeJobs?.groupIds.length)) {
+      const excludedJobAndGroupIds = [
+        ...(excludeJobs?.jobIds ?? []),
+        ...(excludeJobs?.groupIds ?? []),
+      ];
+      const excludedJobsResponse = (
+        await mlClient.getJobs<MlJobsResponse>({
+          job_id: excludedJobAndGroupIds.join(','),
+        })
+      ).body.jobs;
+
+      const excludedJobsIds: Set<string> = new Set(excludedJobsResponse.map((v) => v.job_id));
+
+      resultJobIds = resultJobIds.filter((v) => !excludedJobsIds.has(v));
+    }
+
+    return resultJobIds;
+  };
+
+  return {
+    /**
+     * Gets not started datafeeds for provided jobs selection.
+     * @param includeJobs
+     * @param excludeJobs
+     */
+    async getNotStartedDatafeed(
+      includeJobs: JobSelection,
+      excludeJobs?: JobSelection | null
+    ): Promise<NotStartedDatafeedResponse | void> {
+      const jobIds = await getResultJobIds(includeJobs, excludeJobs);
+      logger.debug(`Performing health checks for job ids: ${jobIds.join(', ')}`);
+      const dataFeeds = await datafeedsService.getDatafeedByJobId(jobIds);
+
+      if (dataFeeds) {
+        const {
+          body: { datafeeds: datafeedsStats },
+        } = await mlClient.getDatafeedStats({
+          datafeed_id: dataFeeds.map((d) => d.datafeed_id).join(','),
+        });
+
+        // match datafeed stats with the job ids
+
+        return (datafeedsStats as DatafeedStats[])
+          .filter((datafeedStat) => datafeedStat.state !== 'started')
+          .map((datafeedStats) => {
+            return {
+              ...datafeedStats,
+              job_id: dataFeeds.find((d) => d.datafeed_id === datafeedStats.datafeed_id)!.job_id,
+            };
+          });
+      }
+    },
+    /**
+     * Retrieves report grouped by test.
+     */
+    async getTestsResults({
+      testsConfig,
+      includeJobs,
+      excludeJobs,
+    }: AnomalyDetectionJobsHealthRuleParams): Promise<TestsResults> {
+      const config = getResultJobsHealthRuleConfig(testsConfig);
+
+      const results: TestsResults = [];
+
+      if (config.dataFeed.enabled) {
+        const response = await this.getNotStartedDatafeed(includeJobs, excludeJobs);
+        if (response && response.length > 0) {
+          results.push({
+            name: 'dataFeed',
+            context: {
+              jobIds: [...new Set(response.map((v) => v.job_id))],
+              message: i18n.translate(
+                'xpack.ml.alertTypes.jobsHealthAlertingRule.dataFeedStateMessage',
```

Comment:
Nit - property should be `datafeedStateMessage`

Reply:
Renamed in https://github.com/elastic/kibana/pull/106084/commits/b08031efbc265847be7d3617ca90253110b74fc9